### PR TITLE
Update CA1054, CA1055 and CA1056 resources

### LINKED ISSUE #3009: CA1054 reports that I should use a Uri type, even though the parameter is clearly named with "Url"
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/804594/ca1054-reports-that-i-should-use-a-uri-type-even-t.html)._

---
URI and URL are not interchangeable things.

This is happening after I added the new FxCop roslyn analyzer package to a project. I have to either configure a bunch of suppressions or disable the CA rule. Neither are optimal.

---
#### Visual Studio Feedback System on 11/3/2019, 10:50 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Visual Studio Feedback System on 11/4/2019, 08:58 AM: 

<p>This issue is currently being investigated. Our team will get back to you if either more information is needed, a workaround is available, or the issue is resolved.</p>

---
### Original Solutions
(no solutions)

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #3009

### CODE REVIEW

**Comment by Evangelink** [comment]
I am wondering if we should update the documentation to actually mention that URL and URN are both subsets or URI, this could make it easier for people.

The linked issue is a user who didn't understand why the rule was suggesting `Uri` when he express `Url`.

File: src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/MicrosoftCodeQualityAnalyzersResources.resx [hunk 1]
```diff
@@ -605,31 +605,31 @@
     <value>Do not declare visible instance fields</value>
   </data>
   <data name="UriParametersShouldNotBeStringsTitle" xml:space="preserve">
-    <value>Uri parameters should not be strings</value>
+    <value>URI-like parameters should not be strings</value>
   </data>
   <data name="UriParametersShouldNotBeStringsDescription" xml:space="preserve">
-    <value>If a method takes a string representation of a URI, a corresponding overload should be provided that takes an instance of the URI class, which provides these services in a safe and secure manner.</value>
+    <value>This rule assumes that the parameter represents a Uniform Resource Identifier (URI). A string representation or a URI is prone to parsing and encoding errors, and can lead to security vulnerabilities. he System.Uri class provides these services in a safe and secure manner.</value>
```
Comment (mavasani) [inline src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/MicrosoftCodeQualityAnalyzersResources.resx:611]:
```suggestion
    <value>This rule assumes that the parameter represents a Uniform Resource Identifier (URI). A string representation or a URI is prone to parsing and encoding errors, and can lead to security vulnerabilities. 'System.Uri' class provides these services in a safe and secure manner.</value>
```

**Comment by mavasani** [comment]
@Evangelink You will need to fix the xlf files too :-)

**Comment by Evangelink** [comment]
Yes! Doing it right now, I always forget xD

**Comment by Evangelink** [comment]
@mavasani ready to be merged

**Pull request merged**